control_protocol.md

```diff
@@ -0,0 +1,359 @@
+The first and fourth frames are empty frames.
+
+
+#### Configuration
+
+Each {ref}`Coordinator <components.md#coordinator>` shall offer one ROUTER socket, bound to a host name (or IP address or any address of a computer with "*") and port.
+
+{ref}`Components <components.md#components>` shall have one DEALER socket connecting to one Coordinator's ROUTER socket.
+
+Coordinators shall have one DEALER socket per other Coordinator in the Network.
+This DEALER socket shall connect to the other Coordinator's ROUTER socket.
+
+Messages must be sent to a Coordinator's ROUTER socket.
+
+
+(router-sockets)=
+#### Particularities of ROUTER sockets
+
+A small introduction to ROUTER sockets, for more details see [zmq guide chapter 3](https://zguide.zeromq.org/docs/chapter3/#Exploring-ROUTER-Sockets).
+
+A router socket assigns a random _identity_ to each connecting peer.
+If, for example, two Components `CA`, `CB` connect to the ROUTER sockets, the socket assigns the identities, which we call `IA`, `IB` (they can be any bytes sequence).
+
+Whenever a message is sent to the ROUTER socket from a peer, it reads as a first frame that identity.
+For example if `CA` sends the message `Request A`, the ROUTER socket will read `IA|Request A`.
+That way, you can return an answer to exactly the original peer and not, for example `CB`.
+If you call the ROUTER's send command with `IA|Reply A`, the socket will send `Reply A` to the peer, whose connection is `IA`, in this case `CA`.
+
+The following diagram shows this example communication with two Components:
+sequenceDiagram
+    participant Code as Message handling
+    participant ROUTER as ROUTER socket
+    Note over Code, ROUTER: Coordinator
+    CA ->> ROUTER: "Request A"
+    ROUTER ->> Code: "IA|Request A"
+    Code ->> ROUTER: "IA|Reply A"
+    ROUTER ->> CA: "Reply A"
+    CB ->> ROUTER: "Request B"
+    ROUTER ->> Code: "IB|Request B"
+    Code ->> ROUTER: "IB|Reply B"
+    ROUTER ->> CB: "Reply B"
+:::
+
+
+### Protocol basics
+
+#### Naming scheme
+
+Each Component has an individual name, given by the user, the _Component Name_.
+A Component Name must be a series of bytes, without the ASCII character "." (byte value 46).
+Component Names must be unique in a {ref}`Node <network-structure.md#node>`, i.e. among the Components connected to a single Coordinator.
+The Coordinator itself has the Component Name `COORDINATOR` (ASCII encoded).
+
+Similarly, every node has a name, the _Namespace_.
+Every Namespace has to be unique in the Network.
+As each Component belongs to exactly one Node, it is fully identified by the combination of Namespace and Component Name, which is globally unique.
+This _full name_ is the composition of Namespace, ".", and Component Name.
+For example `N1.CA` is the full name of the Component `CA` in the Node `N1`.
+
+The receiver of a message may be specified without the Namespace, if the receiver lives in the same Node.
```

Comment:
I'm unsure, whether we should add that single exception to the name definition, or only in the section regarding the sign in process.

Reply:
Done manually. I left it open for the discussion.

Reply:
> I'm unsure, whether we should add that single exception to the name definition, or only in the section regarding the sign in process.

Well, definitely to the name definition here, otherwise the section on sign in might contradict the definition here: _Here we say "must be Full name" - in the sign in section we say "well, except during sign-in"._
We can have it in the sign in section, too, of course.

Reply:
I added the text, as suggested.